packages/aws-cdk-lib/pipelines/lib/codepipeline/codepipeline-source.ts

```diff
@@ -256,11 +256,11 @@ class GitHubSource extends CodePipelineSource {
     super(repoString);
 
     const parts = repoString.split('/');
-    if (Token.isUnresolved(repoString) || parts.length !== 2) {
+    if (Token.isUnresolved(repoString) || parts.length < 2) {
       throw new Error(`GitHub repository name should be a resolved string like '<owner>/<repo>', got '${repoString}'`);
```

Comment:
Lets move this validation into a function. It should cover the edge case of a string such as 'owner/group1//repo' and give a specific error message.